File: APNs/AppKit/AppService.swift
//
//  AppService.swift
//  APNs
//
//  Created by Arror on 2019/3/3.
//  Copyright © 2019 Arror. All rights reserved.
//

import Foundation

public final class AppService {
    
    public let logger = Logger()
    
    public static let current: AppService = AppService()
    
    public let environment: AppEnvironment
    
    private init() {
        self.environment = AppEnvironment(identifier: .production)
        self.logger.add(destination: ConsoleLogDestination(name: "com.APNs.ConsoleLogDestination"))
    }
}

File: flutter/apns/lib/app_kit/app_service.dart
class AppService {

  AppService._();

  AppEnvironment _environment;
  AppEnvironment get environemnt => _environment;
  
  static AppService _shared;

  static AppService get current => _sharedInstance();

  static AppService _sharedInstance() {
    if (_shared == null) {
      _shared = AppService._();
      _shared._environment = AppEnvironment(AppEnvironmentIdentifier.production, AppUser());
    }
    return _shared;
  }  
}

enum AppEnvironmentIdentifier {
  production
}

class AppEnvironment {

  static AppEnvironment get current => AppService.current.environemnt;

  AppEnvironmentIdentifier _identifier;
  AppEnvironmentIdentifier get identifier => _identifier;

  AppUser _user;
  AppUser get user => _user;

  AppEnvironment(this._identifier, this._user);
}

class AppUser {
  
  
}